p4-16/psa/PSA.mdk

```diff
@@ -911,36 +897,41 @@ the cloned copy to `clone_port`.
 Packet resubmission is a mechanism to repeat ingress processing on a
 packet.
 
-If a packet is resubmitted, it re-enters the ingress parser after the
-ingress pipeline processing. The `ingress_port` of the resubmitted
-packet is the same as the original packet, however, the
-`packet_path` of the resubmitted packet is changed to `RESUBMIT`.
-The resubmitted packet has the same header and payload as the original
+Packet resubmission happens at the end of the ingress pipeline. When a
+packet is resubmitted, the packet finishes the ingress pipeline
+processing and re-enters the ingress parser without being deparsed. In
+other words, the resubmitted packet has the same header and payload as
+the original packet. The `ingress_port` of the resubmitted packet is
+the same as the original packet. The `packet_path` of the resubmitted
+packet is changed to `RESUBMIT`.
+
+The ingress parser distinguishes the resubmitted packet from the
+original packet with the `packet_path` field in
+`ingress_parser_intrinsic_metadata_t`. The ingress parser can choose a
+different algorithm to parse the resubmitted packet. Similarly, the
+ingress pipeline can choose to process the resubmitted packet with
+different actions as opposed to the ones used to process the original
```

Comment:
Might be nice to mention that not only can you tell the original packet from a resubmitted packet because of `packet_path`, but the user's program can make it possible to distinguish between a packet resubmitted the first time, or second time, etc. by the extra metadata they can associate with the packet.  As far as I can tell, that is the _only_ way to cause a packet to be processed differently on the 2nd or N-th times being resubmitted.